### PR TITLE
[16.0][MIG] base_business_document_import

### PR DESCRIPTION
**Pull request by ghost**
Depends on:
- [x] #723

### CODE REVIEW

**Comment by StephaneMangin** [comment]
cc @gbrito-erpopen Could you check for your dependencies please ? Once fixed, we could move forward.

**Comment by ghost** [comment]
@StephaneMangin 
There is an open pr for the dependency #723

**Comment by simahawk** [comment]
> @StephaneMangin There is an open pr for the dependency #723

I added it to the description, can you take of this next time? :pray: 
Also, you can fix tests here by including a pending PR by adding a separated commit w/ this change in `tests-requirements.txt`:

```
odoo-addon-pdf_helper @ git+https://github.com/OCA/edi@refs/pull/723/head#subdirectory=setup/pdf_helper
```

**Comment by ghost** [comment]
> > @StephaneMangin There is an open pr for the dependency #723
> 
> I added it to the description, can you take of this next time? pray Also, you can fix tests here by including a pending PR by adding a separated commit w/ this change in `tests-requirements.txt`:
> 
> ```
> odoo-addon-pdf_helper @ git+https://github.com/OCA/edi@refs/pull/723/head#subdirectory=setup/pdf_helper
> ```

Good tip, I will for sure use it

**Comment by ghost** [comment]
@simahawk I added the test-requirements file, the tests pass, except the unreleased dependencies. is there something I can do to make sure it is also green?

**Comment by simahawk** [comment]
> @simahawk I added the test-requirements file, the tests pass, except the unreleased dependencies. is there something I can do to make sure it is also green?

No, that's normal. We cannot merge a PR if it misses another PR to work.
This is also the nice thing of having the pending merge added in tests-requirements.txt:  you get automatic check for it.
TIP: always put such change in a different commit (eg: `Add TMP test dependency`) so that:
1. is clearly visible in the commit list
2. to remove it, you just discard the commit w/o rewriting the last good one

**Comment by ghost** [comment]
Ah, i see i will make the change in this PR, i add it together. Thanks

**Comment by bosd** [comment]
This one could use a bit more test coverage.

File: base_business_document_import/models/business_document_import.py
Comment (simahawk) [inline base_business_document_import/models/business_document_import.py:1373]: this method should be removed

**Comment by ghost** [comment]
> This one could use a bit more test coverage.

Sure, I will add more tests to it

**Review by sbejaoui — APPROVED** [review]

**Review by StephaneMangin — APPROVED** [review]

File: base_business_document_import/models/business_document_import.py
Comment (acsonefho) [inline base_business_document_import/models/business_document_import.py]: Simper way to do it: `record = model.browse(data_dict["id"]).exists()`
(so the next line with `record.create_date` become useless.)

File: base_business_document_import/models/business_document_import.py
Comment (acsonefho) [inline base_business_document_import/models/business_document_import.py]: Better to build domain like this:
```
from odoo.osv import expression  # Move it on the import secion
domain = expression.AND([domain, [("ref", "=", partner_dict["ref"])]])
self.env["res.partner"].search(domain, limit=1, order=order)
```

File: base_business_document_import/models/business_document_import.py
Comment (acsonefho) [inline base_business_document_import/models/business_document_import.py]: Same here, better to merge domain with `expression.AND(...)`

File: base_business_document_import/models/business_document_import.py
Comment (acsonefho) [inline base_business_document_import/models/business_document_import.py]: Same here, better to merge domain with `expression.AND(...)`

File: base_business_document_import/models/business_document_import.py
Comment (acsonefho) [inline base_business_document_import/models/business_document_import.py]: Same here, better to merge domain with `expression.AND(...)`

File: base_business_document_import/models/business_document_import.py
Comment (acsonefho) [inline base_business_document_import/models/business_document_import.py]: Same here, better to merge domain with `expression.AND(...)`

File: base_business_document_import/models/business_document_import.py
Comment (acsonefho) [inline base_business_document_import/models/business_document_import.py]: Same here, better to merge domain with `expression.AND(...)`

File: base_business_document_import/models/business_document_import.py
Comment (acsonefho) [inline base_business_document_import/models/business_document_import.py]: Same here, better to merge domain with `expression.AND(...)`

File: base_business_document_import/models/business_document_import.py
Comment (acsonefho) [inline base_business_document_import/models/business_document_import.py]: Same here, better to merge domain with `expression.AND(...)`

File: base_business_document_import/models/business_document_import.py
Comment (acsonefho) [inline base_business_document_import/models/business_document_import.py]: Same here, better to merge domain with `expression.AND(...)`

File: base_business_document_import/models/business_document_import.py
Comment (acsonefho) [inline base_business_document_import/models/business_document_import.py:429]: Same here for the entire section, better to merge domain with `expression.AND(...)`

File: base_business_document_import/models/business_document_import.py
Comment (acsonefho) [inline base_business_document_import/models/business_document_import.py:665]: Same here, better to merge domain with `expression.AND(...)`

File: base_business_document_import/models/business_document_import.py
Comment (acsonefho) [inline base_business_document_import/models/business_document_import.py]: Same here, better to merge domain with `expression.AND(...)`

File: base_business_document_import/models/business_document_import.py
Comment (acsonefho) [inline base_business_document_import/models/business_document_import.py]: Same here, better to merge domain with `expression.AND(...)`

File: base_business_document_import/models/business_document_import.py
Comment (acsonefho) [inline base_business_document_import/models/business_document_import.py:783]: Useless to specify `[0]`

**Review by acsonefho — CHANGES_REQUESTED** [review]
Some improvements

File: base_business_document_import/models/business_document_import.py
Comment (ghost) [reply base_business_document_import/models/business_document_import.py]: @acsonefho Nice, looks more clear that way

**Comment by ghost** [comment]
@acsonefho Just implemented your feedback changes, can you please review?

File: base_business_document_import/models/business_document_import.py
Comment (acsonefho) [inline base_business_document_import/models/business_document_import.py:139]: IMO Not in favor of this syntax who is less readable.. But why not.

**Review by acsonefho — APPROVED** [review]

**Review by sebalix — APPROVED** [review]

**Comment by acsonefho** [comment]
@OCA/edi-maintainers Seems ready to merge? :pray: Thanks in advance :muscle:

**Review by TDu — APPROVED** [review]

**Comment by acsonefho** [comment]
@rousseldenis Do you have super-power here to merge?

**Comment by rousseldenis** [comment]
> @rousseldenis Do you have super-power here to merge?

Nope sorry. @OCA/edi-maintainers

**Comment by sebalix** [comment]
cc @simahawk this one is ready to merge

**Comment by simahawk** [comment]
checking...

**Review by simahawk — APPROVED** [review]
I guess we can live a bit more w/ a deprecated method...

**Pull request merged**
